From a27a83da30d58511b2a5f2cdb13ebd4fca7a3d0a Mon Sep 17 00:00:00 2001
From: Pavel Grunt <pgrunt@redhat.com>
Date: Tue, 1 Nov 2016 17:08:01 +0100
Subject: [PATCH] Adjust include header to new location of macros
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

minor & major macros were moved to sysmacros.h

usbutil.c: In function ‘spice_usbutil_get_sysfs_attribute’:
usbutil.c:110:14: warning: ‘__major_from_sys_types’ is deprecated:
  In the GNU C Library, `major' is defined by <sys/sysmacros.h>.
  For historical compatibility, it is currently defined by
  <sys/types.h> as well, but we plan to remove this soon.
  To use `major', include <sys/sysmacros.h> directly.
  If you did not intend to use a system-defined macro `major',
  you should #undef it after including <sys/types.h>.
  [-Wdeprecated-declarations]
              major(stat_buf.st_rdev), minor(stat_buf.st_rdev), attribute);

Acked-by: Victor Toso <victortoso@redhat.com>
---
 src/usbutil.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/usbutil.c b/src/usbutil.c
index 7bfbe44..b68a2e1 100644
--- a/src/usbutil.c
+++ b/src/usbutil.c
@@ -30,7 +30,10 @@
 #ifdef __linux__
 #include <stdio.h>
 #include <unistd.h>
+#include <sys/sysmacros.h>
+#ifndef major /* major and minor macros were moved to sys/sysmacros.h from sys/types.h */
 #include <sys/types.h>
+#endif
 #include <sys/stat.h>
 #endif
 #include "usbutil.h"
-- 
2.10.2

